### ISSUE TITLE
预测掩膜大小问题

### ISSUE THREAD

**Issue by I-AM-Not-Xx** [SunnyHaze/IML-ViT#57]
您好，再次叨扰！
在论文中看到您经过MLP输出的预测掩膜尺寸是H/4*W/4*1
但是在代码中发现您是将mask_pred张量调整为input_size,self.input_size是1024*1024吗，请问是以哪个为准呢
代码来自IML-ViT/iml_vit_model.py的最后一段
    def forward(self, x:torch.Tensor, masks, edge_masks, shape= None):
        x = self.encoder_net(x)  # 将输入 x 传入编码器网络
        x = self.featurePyramid_net(x) # 将编码器网络的输出传入特征金字塔网络
        feature_list = [] # 初始化特征列表
        # 遍历特征金字塔网络的输出
        for k, v in x.items(): 
            # 将每个特征添加到特征列表中
            feature_list.append(v)
        # 将特征列表传入预测头部
        x = self.predict_head(feature_list)
        
        # 上采样到 1024x1024
        #torch.nn.functional.interpolate 是 PyTorch 中用于调整张量尺寸的函数，支持多种插值方法（如双线性插值、最近邻插值等）。它的核心作用是将输入张量的尺寸调整为目标尺寸
        # x为输入张量，mode这里选的是双线性插值，align_corners=False表示在插值时不对齐角点
        # size=(self.input_size, self.input_size)表示将输出张量的尺寸
        mask_pred = F.interpolate(x, size = (self.input_size, self.input_size), mode='bilinear', align_corners=False)
        
        # 计算损失
        #计算预测损失
        predict_loss = self.BCE_loss(mask_pred, masks)
        edge_loss = F.binary_cross_entropy_with_logits(
            input = mask_pred,
            target= masks, 
            weight = edge_masks
            ) * self.edge_lambda # 计算边缘损失
        predict_loss += edge_loss # 将边缘损失加到预测损失中
        mask_pred = torch.sigmoid(mask_pred) # 对预测结果应用 sigmoid 激活函数
        
        return predict_loss, mask_pred, edge_loss # 返回预测损失、预测掩码和边缘损失

**Comment by SunnyHaze** [comment]
不矛盾的。

论文中和代码中，模型实际经过神经网络输出的尺寸就是[H/4, W/4]，这主要是为了避免神经网络输出过多pixel带来巨大的计算开销。当网络输出pred_mask后，上采样是一个开销很小的操作，可以看做为了计算loss的单独步骤。主要用于对齐输入图片的尺寸。
